includes/admin/upgrades/upgrade-functions.php

```diff
@@ -1328,498 +1328,227 @@ function edd_include_file_download_log_migration_batch_processor( $class = '' )
 
-	$customer_email_addresses_complete = edd_has_upgrade_completed( 'migrate_customer_email_addresses' );
-
-	/** Logs Migration *******************************************************/
-	$logs = $wpdb->get_results( "SELECT t.slug
-								 FROM {$wpdb->posts} AS p
-								 LEFT JOIN {$wpdb->term_relationships} AS tr ON (p.ID = tr.object_id)
-								 LEFT JOIN {$wpdb->term_taxonomy} AS tt ON (tr.term_taxonomy_id = tt.term_taxonomy_id)
-								 LEFT JOIN {$wpdb->terms} AS t ON (tt.term_id = t.term_id)
-								 WHERE p.post_type = 'edd_log' AND t.slug != 'sale'
-								 GROUP BY p.ID
-								 LIMIT 1" );
-	$logs_complete = edd_has_upgrade_completed( 'migrate_logs' );
-
-	/** Order Notes Migration ************************************************/
-	$order_notes_complete         = edd_has_upgrade_completed( 'migrate_order_notes' );
-	$order_notes_removal_complete = edd_has_upgrade_completed( 'remove_legacy_order_notes' );
-
-	/** Customer Notes Migration *********************************************/
-	$customer_notes_complete         = edd_has_upgrade_completed( 'migrate_customer_notes' );
-	$customer_notes_removal_complete = edd_has_upgrade_completed( 'remove_legacy_customer_notes' );
-
-	// Initialise to first step.
-	$step = 1; ?>
+	/*
+	 * Determine if legacy data can be removed.
+	 * It can be if all upgrades except legacy data have been completed.
+	 */
+	$can_remove_legacy_data = array_key_exists( 'v30_legacy_data_removed', $upgrade_statuses ) && $upgrade_statuses[ 'v30_legacy_data_removed' ] !== true;
+	if ( $can_remove_legacy_data ) {
+		foreach( $upgrade_statuses as $upgrade_key => $status ) {
+			if ( 'v30_legacy_data_removed' === $upgrade_key ) {
+				continue;
+			}
 
-	<div id="edd-migration-ready" class="notice notice-success" style="display: none;">
-		<p>
-			<?php echo wp_kses( __( '<strong>Database Upgrade Complete:</strong> All database upgrades have been completed.', 'easy-digital-downloads' ), array( 'strong' => array() ) ); ?>
-			<br /><br />
-			<?php esc_html_e( 'You may now leave this page.', 'easy-digital-downloads' ); ?>
-		</p>
-	</div>
+			// If there's at least one upgrade still unfinished, we can't remove legacy data.
+			if ( true !== $status ) {
+				$can_remove_legacy_data = false;
+				break;
+			}
+		}
+	}
 
+	if ( $migration_complete ) {
+		?>
+		<div id="edd-migration-ready" class="notice notice-success">
+			<p>
+				<?php echo wp_kses( __( '<strong>Database Upgrade Complete:</strong> All database upgrades have been completed.', 'easy-digital-downloads' ), array( 'strong' => array() ) ); ?>
+				<br /><br />
+				<?php esc_html_e( 'You may now leave this page.', 'easy-digital-downloads' ); ?>
+			</p>
+		</div>
+		<?php
+		return;
+	}
+	?>
 	<div id="edd-migration-nav-warn" class="notice notice-info">
 		<p><?php echo wp_kses( __( '<strong>Important:</strong> Do not navigate away from this page until all upgrades complete.', 'easy-digital-downloads' ), array( 'strong' => array() ) ); ?></p>
```

Comment:
Although I think this is technically correct wording, it feels a bit grammatically stilted, maybe? Could we change it to something like:

> **Important:** Do not navigate away from this page until all upgrades have completed.